src/goto-symex/symex_builtin_functions.cpp

```diff
@@ -45,8 +45,7 @@ void goto_symext::symex_allocate(
   const exprt &lhs,
   const side_effect_exprt &code)
 {
-  if(code.operands().size()!=2)
-    throw "allocate expected to have two operands";
+  PRECONDITION(code.operands().size() == 2);
```

Comment:
DATA_INVARIANT?

Reply:
No, for the same reason as above (asserts properties of function input).